src/main/java/com/hubspot/jinjava/lib/filter/JoinFilter.java

```diff
@@ -52,17 +56,36 @@ public Object filter(Object var, JinjavaInterpreter interpreter, String... args)
     }
 
     ForLoop loop = ObjectIterator.getLoop(var);
+    boolean first = true;
     while (loop.hasNext()) {
       Object val = loop.next();
 
       if (attr != null) {
         val = interpreter.resolveProperty(val, attr);
       }
 
-      vals.add(Objects.toString(val, ""));
+      try {
```

Comment:
You can also simple sum the length of each string and compare to the limit. It may look simpler. Up to you.

Reply:
That's how the modified string builder works and throws the exception once the length > limit. I didn't see the need to duplicate the code we already have in there.